docs/feature_oled_driver.md

```diff
@@ -72,6 +72,43 @@ static void render_logo(void) {
 }
 ```
 
+## Buffer Read Example
+For some purposes, you may need to read the current state of the OLED display
+buffer. The `oled_read_raw` function can be used to safely read bytes from the
+buffer.
+
+In this example, calling `fade_display` in the `oled_task_user` function will
+slowly fade away whatever is on the screen by turning random pixels black over
+time.
+```c
+//Setup some mask which can be or'd with bytes to turn off pixels
+const uint8_t single_bit_masks[8] = {127, 191, 223, 239, 247, 251, 253, 254};
+
+static void fade_display(void) {
+    //Define the reader structure
+    oled_buffer_reader_t reader;
+    uint8_t buff_char;
+    if(random() % 30==0) {
+	srand(timer_read());
+	// Fetch a pointer for the buffer byte at index 0. The return structure
+	// will have the pointer and the number of bytes remaining from this
+	// index position if we want to perform a sequential read by
+	// incrementing the buffer pointer
+	reader = oled_read_raw(0);
+	//Loop over the remaining buffer and erase pixels as we go
+        for (uint16_t i = 0; i < reader.remaining_element_count; i++) {
+	    //Get the actual byte in the buffer by dereferencing the pointer
+            buff_char = *reader.current_element;
+	    if (buff_char != 0) {
+                oled_write_raw_byte(buff_char & single_bit_masks[rand() % 8], i);
+	    }
+	    //increment the pointer to fetch a new byte during the next loop
+	    reader.current_element++;
+        }
+    }
+}
+```
```

Comment:
Can we get the tabs/spaces fixed up here?

Reply:
```suggestion
static void fade_display(void) {
    //Define the reader structure
    oled_buffer_reader_t reader;
    uint8_t buff_char;
    if(random() % 30==0) {
    srand(timer_read());
    // Fetch a pointer for the buffer byte at index 0. The return structure
    // will have the pointer and the number of bytes remaining from this
    // index position if we want to perform a sequential read by
    // incrementing the buffer pointer
    reader = oled_read_raw(0);
    //Loop over the remaining buffer and erase pixels as we go
        for (uint16_t i = 0; i < reader.remaining_element_count; i++) {
        //Get the actual byte in the buffer by dereferencing the pointer
            buff_char = *reader.current_element;
        if (buff_char != 0) {
                oled_write_raw_byte(buff_char & single_bit_masks[rand() % 8], i);
        }
        //increment the pointer to fetch a new byte during the next loop
        reader.current_element++;
        }
    }
}
```
```

Reply:
Yes, very sorry about the tab/space issue here.   I've cleaned this up and checked it via the GitHub markdown renderer.

Thank you for the review!